internal/pkg/cli/job_deploy.go

```diff
@@ -0,0 +1,131 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"fmt"
+
+	"github.com/aws/copilot-cli/internal/pkg/aws/sessions"
+	"github.com/aws/copilot-cli/internal/pkg/config"
+	"github.com/aws/copilot-cli/internal/pkg/manifest"
+	"github.com/aws/copilot-cli/internal/pkg/term/command"
+	termprogress "github.com/aws/copilot-cli/internal/pkg/term/progress"
+	"github.com/aws/copilot-cli/internal/pkg/term/selector"
+	"github.com/aws/copilot-cli/internal/pkg/workspace"
+	"github.com/spf13/cobra"
+)
+
+type deployJobVars struct {
+	*GlobalOpts
+	Name         string
+	EnvName      string
+	ImageTag     string
+	ResourceTags map[string]string
+}
+
+type deployJobOpts struct {
+	deployJobVars
+
+	store        store
+	ws           wsSvcDirReader
+	unmarshal    func(in []byte) (interface{}, error)
+	cmd          runner
+	sessProvider sessionProvider
+
+	spinner progress
+	sel     wsSelector
+
+	// imageBuilderPusher imageBuilderPusher
+	// s3           artifactUploader
+	// addons       templater
+	// appCFN       appResourcesGetter
+	// jobCFN       cloudformation.CloudFormation
+
+	// // cached variables
+	// targetApp         *config.Application
+	// targetEnvironment *config.Environment
+	// targetWorkload    *config.Service
+}
+
+func newJobDeployOpts(vars deployJobVars) (*deployJobOpts, error) {
+	store, err := config.NewStore()
+	if err != nil {
+		return nil, fmt.Errorf("new config store: %w", err)
+	}
+
+	ws, err := workspace.New()
+	if err != nil {
+		return nil, fmt.Errorf("new workspace: %w", err)
+	}
+	return &deployJobOpts{
+		deployJobVars: vars,
+
+		store:        store,
+		ws:           ws,
+		unmarshal:    manifest.UnmarshalService,
```

Comment:
Should we have a different unmarshal func for job manifest? If so maybe defining a stub one for job?

Reply:
This is going to get renamed to "UnmarshalWorkload" in a future PR.

Reply:
I realized midway through that "UnmarshalService" just uses a switch statement on the `type` field in the manifest, which means we can just add new cases as necessary to generate the appropriate job struct. I'm keeping this as is and renaming `manifest.UnmarshalService` to `manifest.UnmarshalWorkload`

Reply:
Yeah feel free to do the renaming in the future PR since we might need to rename `svc.go` to `workload.go` in manifest pkg i assume?

Reply:
yeah, I'll do this in a separate PR. I still need to update the `config` and `manifest` packages for this rename.